drivers/encx24j600/encx24j600.c

```diff
@@ -360,7 +360,7 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
     /* hdr.frame_len given by device contains 4 bytes checksum */
     size_t payload_len = hdr.frame_len - 4;
 
-    if (buf) {
+    if (buf && (len >= payload_len)) {
```

Comment:
did you hit this somehow? the correct use case would be to call recv () first without buf to get the required size, then again with a buffer large enough. is there any place where this procedure is not followed?